### PR TITLE
[#19921] feat: remove key pair action

### LINKED ISSUE #19921: Allow user to remove key-pairs and derived accounts while inside wallet settings
# Feature Issue

### User Story

As a user,
I want remove a key-pair while inside the wallet settings.

### Description 
Implement the remove key-pair feature for the wallet settings.

### Designs

Figma flow: [the Figma designs](https://www.figma.com/file/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?type=design&node-id=56-102822&mode=design&t=M9zZRtqh1MBl3m5j-4)

### Notes
* Based on [the design notes here](https://www.figma.com/file/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?type=design&node-id=57-103715&mode=design&t=M9zZRtqh1MBl3m5j-4), the user should not be allowed to remove the default key-pair.

### Acceptance Criteria

| Screen | Description | Acceptance Criteria |
|--------|--------|--------|
| Key-pair Actions Menu | Users can choose to remove a key-pair with actions menu | A user should be able to open the confirmation UI for removing a key-pair. |
| Key-pair Confirm Removal | Users can confirm the removal of a key-pair | A user should be able to swipe to confirm the removal of a key-pair. And the user should be required to authorize this action. |
| Key-pair Removed Toast | A toast will appear to confirm that a key-pair has been removed | The user should see a toast that confirms a key-pair has been removed |
| Key-pair Removed | Users can see the updated list of key-pairs and accounts | The user should see the confirmation toast disappear and the updated list of key-pairs without the removed key-pair. |

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #19921

### Summary

Implement the remove key-pair feature for the wallet settings.

#### Areas that maybe impacted

- Keypairs and account screen
- Remove key pair

### Steps to test

1. Open Wallet setting
2. Click on not default keypair's action button
3. Click on remove key pair

### Result

https://github.com/status-im/status-mobile/assets/71308738/a9ed696e-88d4-46b4-9596-084508c2d2cc

status: ready

### CODE REVIEW

**Comment by mohsen-ghafouri** [comment]
blocked by this issue https://github.com/status-im/status-mobile/issues/19472, when i want to remove a key pair it throws error `no key for given address or file`.

will re-test after https://github.com/status-im/status-mobile/pull/19531

File: src/status_im/contexts/settings/wallet/keypairs_and_accounts/actions/view.cljs [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+                 :content (fn []
+                            [remove-key-pair/view key-pair])}]))
 
 (defn on-show-qr
   [data]
   (rf/dispatch [:open-modal :screen/settings.encrypted-key-pair-qr data]))
 
 (defn view
-  [props data]
+  [props key-pair]
```
Comment (seanstrom) [inline src/status_im/contexts/settings/wallet/keypairs_and_accounts/actions/view.cljs:23]: Nice 😸

File: src/quo/components/text_combinations/page_top/style.cljs [hunk 2]
```diff
@@ -6,10 +6,11 @@
    :padding-horizontal 20
    :row-gap            8})
 
-(def header
+(defn header
+  [title-size]
   {:flex-direction  :row
    :justify-content :space-between
-   :height          32})
+   :height          (if title-size :auto 32)})
```
Comment (seanstrom) [inline src/quo/components/text_combinations/page_top/style.cljs]: Do we know why `:heading-1` would always be height `32`, is that the max-height or something?

**Review by seanstrom — APPROVED** [review]
Nice stuff ✅

**Comment by seanstrom** [comment]
@mohsen-ghafouri One thing I noticed is that the removed-keypair toast should be in light mode 🌤️

File: src/quo/components/text_combinations/page_top/style.cljs (hunk 2 above)
Comment (mohsen-ghafouri) [reply src/quo/components/text_combinations/page_top/style.cljs]: No idea, just wanted to add possibility to change it from parent without breaking any other usages.

File: src/status_im/contexts/settings/wallet/events.cljs [hunk 3]
```diff
@@ -8,17 +8,21 @@
     [utils.security.core :as security]
     [utils.transforms :as transforms]))
 
+(defn- update-keypair
+  [keypairs key-uid update-fn]
+  (mapcat (fn [keypair]
+            (if (= (keypair :key-uid) key-uid)
+              (let [updated (update-fn keypair)]
+                (if (nil? updated) [] [updated]))
```
Comment (BalogunofAfrica) [inline src/status_im/contexts/settings/wallet/events.cljs]: nit: the nested `let` and `nil?` check can be simplified with `if-let`

```suggestion
              (if-let [updated (update-fn keypair)]
                [updated]
                [])
```

**Comment by mohsen-ghafouri** [comment]
@seanstrom good catch, I noticed they removed theme from toast's props. resolved.

**Review by BalogunofAfrica — APPROVED** [review]
LGTM! ✅

File: src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+(defn view
+  [{:keys [name key-uid]}]
+  (let [on-remove (rn/use-callback #(rf/dispatch
+                                     [:wallet/remove-keypair key-uid])
+                                   [key-uid])]
+    [:<>
+     [quo/page-top
+      {:container-style style/header-container
+       :title           (i18n/label :t/remove-key-pair-and-derived-accounts)
+       :title-size      :heading-2
```
Comment (smohamedjavid) [inline src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs]: If the style of the component in the design system and the screen differs, it's best to check with the Design team to confirm it.

File: src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+       :description     :context-tag
+       :context-tag     {:type    :icon
+                         :size    24
+                         :context name
+                         :icon    :i/seed-phrase}}]
```
Comment (smohamedjavid) [inline src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs]: @mohsen-ghafouri - We need to use the `drawer-top` component here. That has the `heading-2` text size.

**Review by smohamedjavid — COMMENTED** [review]
Looks good overall. Nice work 🚀 🙌 @mohsen-ghafouri 

Just a component change in the confirmation sheet.

File: src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs (hunk 5 above)
Comment (mohsen-ghafouri) [reply src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs]: I think `drawer-top` doesn't support icon, i will update it and use it here

Comment (mohsen-ghafouri) [reply src/status_im/contexts/settings/wallet/keypairs_and_accounts/remove/view.cljs]: @smohamedjavid thanks for pointing it out, I replaced

**Comment by mohsen-ghafouri** [comment]
@status-im/mobile-qa could you please check the test results? I skip manual QA as this action is still behind the feature flag

**Review by smohamedjavid — APPROVED** [review]
🚀

**Comment by churik** [comment]
@mohsen-ghafouri sorry for delay, ready to go

**Pull request merged**